modules/core/src/viewports/web-mercator-viewport.js

```diff
@@ -69,14 +71,30 @@ export default class WebMercatorViewport extends Viewport {
     // TODO - just throw an Error instead?
     altitude = Math.max(0.75, altitude);
 
-    const {fov, aspect, focalDistance, near, far} = getProjectionParameters({
-      width,
-      height,
-      pitch,
-      altitude,
-      nearZMultiplier,
-      farZMultiplier
-    });
+    let fovy;
+    let projectionParameters;
+    if (projectionMatrix) {
+      fovy = altitudeToFovy(projectionMatrix[5] / 2);
+      projectionParameters = {focalDistance: altitude, projectionMatrix};
+    } else {
+      fovy = altitudeToFovy(altitude);
+      const {aspect, fov: fovyRadians, near, far} = getProjectionParameters({
```

Comment:
Perf: you can avoid destructing/constructing objects here:

```js
let projectionParameters = null;
if (projectionMatrix) {
  fovy = altitudeToFovy(projectionMatrix[5] / 2);
} else {
  fovy = altitudeToFovy(altitude);
  projectionParameters = getProjectionParameters({...});
}

super({
  // prev opts
  ...projectionParameters,
  focalDistance: altitude
});
```

Note that `orthographic` and `projectionMatrix` are already passed by `...opts`.

Reply:
Fixed & removed `orthographic`